server/src/db/catalog.rs

```diff
@@ -610,23 +621,21 @@ mod tests {
         create_open_chunk(&p1, "table2");
         create_open_chunk(&p2, "table2");
 
-        let a = catalog.filtered_chunks(&Predicate::default(), |_| ());
+        let a = catalog.filtered_chunks(None, None, |_| ());
 
-        let b = catalog.filtered_chunks(&PredicateBuilder::new().table("table1").build(), |_| ());
+        let b = catalog.filtered_chunks(None, Some(&make_set("table1")), |_| ());
 
-        let c = catalog.filtered_chunks(&PredicateBuilder::new().table("table2").build(), |_| ());
+        let c = catalog.filtered_chunks(None, Some(&make_set("table2")), |_| ());
 
-        let d = catalog.filtered_chunks(
-            &PredicateBuilder::new()
-                .table("table2")
-                .partition_key("p2")
-                .build(),
-            |_| (),
-        );
+        let d = catalog.filtered_chunks(Some("p2"), Some(&make_set("table2")), |_| ());
 
         assert_eq!(a.len(), 3);
         assert_eq!(b.len(), 1);
         assert_eq!(c.len(), 2);
         assert_eq!(d.len(), 1);
     }
+
+    fn make_set(s: impl Into<String>) -> BTreeSet<String> {
+        std::iter::once(s.into()).collect::<BTreeSet<_>>()
```

Comment:
This can be:

```suggestion
        std::iter::once(s.into()).collect()
```

Rust can figure out the type to collect from the return type.